pdns/recpacketcache.cc

```diff
@@ -60,10 +63,20 @@ bool RecursorPacketCache::checkResponseMatches(std::pair<packetCache_t::index<Ha
 
     if (now < iter->d_ttd) { // it is right, it is fresh!
       *age = static_cast<uint32_t>(now - iter->d_creation);
+      // we know ttl is > 0
+      uint32_t ttl = static_cast<uint32_t>(iter->d_ttd - now);
+      if (s_refresh_ttlperc > 0) {
+        const uint32_t deadline = iter->getOrigTTL() * s_refresh_ttlperc / 100;
```

Comment:
could we win a few CPU cycles by checking `d_submitted` before doing the rest of the checks?

Reply:
yes